src/python/pants/engine/target_test.py

```diff
@@ -456,16 +456,9 @@ class NoSourcesTgt(Target):
         core_fields = (Tags,)
 
     no_sources_tgt = NoSourcesTgt({Tags.alias: ["demo"]}, address=Address.parse("//:no_sources"))
```

Comment:
`NoSourcesTgt` is a misnomer now. I think we should continue to test the behavior of having no `Sources` field, in addition to this new test for missing `Dependencies`.